beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -3193,15 +3211,27 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         // signed. If we miss the cache or we're producing a block that conflicts with the head,
         // fall back to getting the head from `slot - 1`.
         let state_load_timer = metrics::start_timer(&metrics::BLOCK_PRODUCTION_STATE_LOAD_TIMES);
+
         // Atomically read some values from the head whilst avoiding holding cached head `Arc` any
         // longer than necessary.
         let (head_slot, head_block_root) = {
             let head = self.canonical_head.cached_head();
             (head.head_slot(), head.head_block_root())
         };
         let (state, state_root_opt) = if head_slot < slot {
+            // Attempt an aggressive re-org if configured and the conditions are right.
+            if let Some(re_org_state) = self.get_state_for_re_org(slot, head_slot, head_block_root)
+            {
+                info!(
+                    self.log,
+                    "Proposing block to re-org current head";
+                    "slot" => slot,
+                    "head" => %head_block_root,
```

Comment:
Yeah, agree. We actually log two info logs when we re-org. I've tried to tweak them to be a bit more clear & consistent:

```rust
info!(
    self.log,
    "Attempting re-org due to weak head";
    "head" => ?canonical_head,
    "parent" => ?re_org_head,
    "head_weight" => ?proposer_head.canonical_head_weight,
    "threshold_weight" => ?proposer_head.re_org_weight_threshold,
);
info!(
    self.log,
    "Proposing block to re-org current head";
    "slot" => slot,
    "head_to_reorg" => %head_block_root,
);
```

Unfortunately we can't log the block root of the new head at this point because it hasn't yet been determined.